### PR TITLE
Genotype validation

### PR DESCRIPTION
**Pull request by tcezard**
This PR add extra checks to the genotype validation by comparing the sample to it's neighbour in the pate it arrived in, the plate it was genotyped on and the plate it was sequenced on.

### CODE REVIEW

File: tests/test_quality_control/test_genotype_validation.py [hunk 1]
```diff
@@ -84,25 +85,60 @@ def test__snp_calling(self, mocked_execute):
...
+    def test__merge_validation_results(self):
+        sample2genotype_validation={"T00001P001A01":os.path.join(self.assets_path, 'sample_data', "T00001P001A01-validation.txt"),
+                                    "T00001P001A02":os.path.join(self.assets_path, 'sample_data', "T00001P001A02-validation.txt")}
+        open_file1 = open(sample2genotype_validation.get("T00001P001A01"))
+        open_file2 = open(sample2genotype_validation.get("T00001P001A02"))
+        with patch.object(builtins, 'open') as mocked_open:
+            mock_open_write = MagicMock()
+            mocked_open.side_effect= [open_file1, open_file2, mock_open_write]
+            self.validator._merge_validation_results(sample2genotype_validation)
+            #print(mocked_open.mock_calls)
```
Comment (katieemelianova) [inline tests/test_quality_control/test_genotype_validation.py]: No assert in this test

File: analysis_driver/report_generation/report_crawlers.py [hunk 2]
```diff
@@ -244,8 +256,9 @@ def _populate_lib_info(self, sample_dir):
             ELEMENT_PROJECT_ID: self.project_id,
             ELEMENT_SAMPLE_EXTERNAL_ID: external_sample_name
         }
-
-        bamtools_path = util.find_file(sample_dir, 'bamtools_stats.txt')
+        print(sample_dir)
+        bamtools_path = self.search_file(sample_dir, 'bamtools_stats.txt')
+        print(bamtools_path)
```
Comment (katieemelianova) [inline analysis_driver/report_generation/report_crawlers.py]: remove print statements

File: bin/run_qc.py [hunk 3]
```diff
@@ -44,6 +46,18 @@ def _parse_args():
 
 
 def run_genotype_validation(args):
+    def retrive_data(paths, work_dir, allow_fail=False):
```
Comment (katieemelianova) [inline bin/run_qc.py]: *retrieve_data

File: bin/run_qc.py [hunk 4]
```diff
@@ -51,37 +65,46 @@ def run_genotype_validation(args):
     os.makedirs(work_dir, exist_ok=True)
     # Hack to retrive the fastq file from the CIFS share
```
Comment (katieemelianova) [inline bin/run_qc.py]: Hack to *retrieve

File: tests/test_quality_control/test_genotype_validation.py [hunk 5]
```diff
@@ -84,25 +87,79 @@ def test__snp_calling(self, mocked_execute):
...
         assert mocked_execute.call_count == 1
-        mocked_execute.assert_called_once_with(
-            [command],
-            job_name='genotype_concordance',
-            working_dir=os.path.join(cfg['jobs_dir'], self.sample_id),
-            cpus=4,
-            mem=8,
-            log_commands=False
-        )
+        mocked_execute.assert_called_once_with([command], job_name='genotype_rename', working_dir=work_dir, cpus=4, mem=8, log_commands=False)
```
Comment (katieemelianova) [inline tests/test_quality_control/test_genotype_validation.py:161]: These assert the same thing

**Comment by tcezard** [comment]
close #99

**Pull request merged**